File: dashboard/ios/AttobotDashboard/Views/Screens/AgentsScreen.swift
import SwiftUI

/// Agents table (no poll; uses the shared cached AgentsStore). Row → Messages(agentId:).
struct AgentsScreen: View {
    let navigate: (Route) -> Void
    @State private var res = Resource<[AgentRow]>()

    var body: some View {
        ScreenScroll {
            Text("Agents")
                .font(.system(size: 20, weight: .bold))
                .foregroundStyle(Theme.text)
                .padding(.bottom, 12)

            if let err = res.error, res.value == nil {
                ErrorState(message: err)
            } else if res.isLoading && res.value == nil {
                Spinner()
            } else {
                DataTable(
                    columns: [
                        DataTableColumn<AgentRow>(id: "slug", title: "slug", width: 120) { a in
                            Text(a.slug).font(.system(size: 13, weight: .bold)).foregroundStyle(Theme.accent).lineLimit(1)
                        },
                        DataTableColumn<AgentRow>(id: "enabled", title: "on", width: 40) { a in
                            Text(a.enabled ? "●" : "○").font(.system(size: 13)).foregroundStyle(a.enabled ? Theme.ok : Theme.err)
                        },
                        DataTableColumn<AgentRow>(id: "model", title: "model", width: 150) { a in
                            VStack(alignment: .leading, spacing: 0) {
                                Text(a.modelName ?? "—").font(.system(size: 13)).foregroundStyle(Theme.text).lineLimit(2)
                                Text(a.apiBase ?? "").font(.system(size: 11)).foregroundStyle(Theme.muted).lineLimit(1)
                            }
                        },
                        DataTableColumn<AgentRow>(id: "max_turn", title: "max turn", width: 80) { a in
                            Text("\(a.maxTurn)").font(.system(size: 13)).foregroundStyle(Theme.text)
                        },
                        DataTableColumn<AgentRow>(id: "ctx", title: "ctx tokens", width: 100) { a in
                            Text(a.contextTokens.map { "\($0)" } ?? "—").font(.system(size: 13)).foregroundStyle(Theme.text)
                        },
                        DataTableColumn<AgentRow>(id: "temp", title: "temp / effort", width: 120) { a in
                            Text("\(a.temperature ?? "—") / \(a.reasoningEffort ?? "—")").font(.system(size: 13)).foregroundStyle(Theme.text)
                        },
                        DataTableColumn<AgentRow>(id: "msg", title: "messages", width: 90) { a in
                            Text(a.msgCount).font(.system(size: 13)).foregroundStyle(Theme.text)
                        },
                        DataTableColumn<AgentRow>(id: "mem", title: "memory", width: 80) { a in
                            Text(a.memCount).font(.system(size: 13)).foregroundStyle(Theme.text)
                        },
                        DataTableColumn<AgentRow>(id: "wf", title: "workflows", width: 90) { a in
                            Text(a.wfCount).font(.system(size: 13)).foregroundStyle(Theme.text)
                        },
                        DataTableColumn<AgentRow>(id: "updated", title: "updated", width: 90) { a in
                            Text(Format.timeAgo(a.updatedAt)).font(.system(size: 13)).foregroundStyle(Theme.text)
                        },
                    ],
                    rows: res.value ?? [],
                    onRow: { a in navigate(.messages(agentId: String(a.id))) }
                )
            }
        }
        .navigationTitle("Agents")
        .navigationBarTitleDisplayMode(.inline)
        .task {
            res.start(load: { try await APIClient.getRows("/api/agents") })
        }
    }
}

File: dashboard/android/app/src/main/java/com/attobot/dashboard/ui/screens/AgentsScreen.kt
package com.attobot.dashboard.ui.screens

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import androidx.lifecycle.viewmodel.compose.viewModel
import androidx.navigation.NavHostController
import com.attobot.dashboard.core.AgentRow
import com.attobot.dashboard.core.timeAgo
import com.attobot.dashboard.state.AgentsViewModel
import com.attobot.dashboard.state.UiState
import com.attobot.dashboard.ui.components.DataTable
import com.attobot.dashboard.ui.components.ErrorState
import com.attobot.dashboard.ui.components.LoadingView
import com.attobot.dashboard.ui.components.PullRefreshScreen
import com.attobot.dashboard.ui.components.TableColumn
import com.attobot.dashboard.ui.nav.Routes
import com.attobot.dashboard.ui.theme.accent
import com.attobot.dashboard.ui.theme.err
import com.attobot.dashboard.ui.theme.muted
import com.attobot.dashboard.ui.theme.ok
import com.attobot.dashboard.ui.theme.text

@Composable
fun AgentsScreen(navController: NavHostController) {
    val vm: AgentsViewModel = viewModel()
    val state by vm.state.collectAsStateWithLifecycle()
    val s = state

    PullRefreshScreen(refreshing = false, onRefresh = null) {
        Text(
            "Agents",
            color = text,
            fontSize = 20.sp,
            fontWeight = FontWeight.Bold,
            modifier = Modifier.padding(bottom = 12.dp),
        )
        when (s) {
            is UiState.Loading -> LoadingView()
            is UiState.Error -> ErrorState(s.message)
            is UiState.Ready -> DataTable(
                rows = s.data,
                onRow = { navController.navigate(Routes.withAgent("messages", it.id.toString())) },
                columns = agentColumns(),
            )
        }
    }
}

private fun agentColumns(): List<TableColumn<AgentRow>> = listOf(
    TableColumn(header = "slug", width = 120.dp) {
        Text(it.slug, color = accent, fontWeight = FontWeight.Bold, fontSize = 13.sp, maxLines = 1)
    },
    TableColumn(header = "on", width = 40.dp) {
        Text(if (it.enabled) "●" else "○", color = if (it.enabled) ok else err, fontSize = 13.sp)
    },
    TableColumn(header = "model", width = 150.dp) {
        Column {
            Text(it.modelName ?: "—", color = text, fontSize = 13.sp, maxLines = 2)
            Text(it.apiBase ?: "", color = muted, fontSize = 11.sp, maxLines = 1)
        }
    },
    TableColumn(header = "max turn", width = 80.dp) {
        Text("${it.maxTurn}", color = text, fontSize = 13.sp)
    },
    TableColumn(header = "ctx tokens", width = 100.dp) {
        Text(it.contextTokens?.toString() ?: "—", color = text, fontSize = 13.sp)
    },
    TableColumn(header = "temp / effort", width = 120.dp) {
        Text("${it.temperature ?: "—"} / ${it.reasoningEffort ?: "—"}", color = text, fontSize = 13.sp)
    },
    TableColumn(header = "messages", width = 90.dp) {
        Text(it.msgCount, color = text, fontSize = 13.sp)
    },
    TableColumn(header = "memory", width = 80.dp) {
        Text(it.memCount, color = text, fontSize = 13.sp)
    },
    TableColumn(header = "workflows", width = 90.dp) {
        Text(it.wfCount, color = text, fontSize = 13.sp)
    },
    TableColumn(header = "updated", width = 90.dp) {
        Text(timeAgo(it.updatedAt), color = text, fontSize = 13.sp)
    },
)
